File: Domain/Company.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Company
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public long? PhoneNumber { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: client-app/src/app/models/company.ts
export class CompanyFormValues {
    id?: string = undefined;
    name: string = '';
    constructor(company?: CompanyFormValues) {
        if (company) {
            this.id = company.id;
            this.name = company.name;
        }
    }
}

export class Company {
    id: string = '';
    name: string = '';
    constructor(company?: Company) {
        if (company) {
            this.id = company.id;
            this.name = company.name;
        }
    }
}